### PR TITLE
Added device Samsung Galaxy Note 2 (N7100) support

### PR DESCRIPTION
**Pull request by JamesHealdUK**

### CODE REVIEW

File: aports/device-samsung-n7100/deviceinfo [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+deviceinfo_dtb=""
+deviceinfo_modules_initfs=""
+deviceinfo_external_disk="true"
+deviceinfo_external_disk_install="true"
+deviceinfo_flash_methods="heimdall-bootimg"
+deviceinfo_arch="armhf"
+
+# Heimdall related
+deviceinfo_flash_heimdall_partition_kernel="BOOT"
+deviceinfo_weston_core_modules="xwayland.so"
```
Comment (drebrez) [inline aports/device-samsung-n7100/deviceinfo]: @JamesHealdUK are these not necessary?
```
deviceinfo_generate_bootimg="true"
deviceinfo_flash_offset_base="0x......"
deviceinfo_flash_offset_kernel="0x...."
deviceinfo_flash_offset_ramdisk="0x...."
deviceinfo_flash_offset_second="0x...."
deviceinfo_flash_offset_tags="0x....."
deviceinfo_flash_cmdline=""
deviceinfo_flash_pagesize="2048"
```

Comment (JamesHealdUK) [inline aports/device-samsung-n7100/deviceinfo]: oh yeah forgot to let you know about that. I just started flashing before you sent me those, I was about to put them in and my phone rebooted into pmOS... It seems everything works without those.

Comment (JamesHealdUK) [inline aports/device-samsung-n7100/deviceinfo]: Have just tested this, did ./pmbootstrap zap and then rebuilt everything and flashed, boots fine! Like you say, thats really odd considering my deviceinfo has no offsets or even a generate_bootimg attribute.

Dunno what you want to do about that, the balls in your court.

**Comment by drebrez** [comment]
I just checked out the branch and the checksums for the `device-samsung-n7100` package are wrong, please run `pmbootstrap checksum device-samsung-n7100` .

And I can also confirm that the boot.img doesn't get generated without the `deviceinfo_generate_bootimg="true"` in the deviceinfo, and trying to run `pmbootstrap flasher flash_kernel` I get this error:
```
[00:46:32] (rootfs_samsung-n7100) mkinitfs samsung-n7100
[00:46:37] (native) flash kernel samsung-n7100
< wait for any device >
ERROR: Failed to open file "/mnt/rootfs_samsung-n7100/boot/boot.img-samsung-n7100"
[00:47:29] ERROR: Command failed: (native) % heimdall flash --BOOT /mnt/rootfs_samsung-n7100/boot/boot.img-samsung-n7100
[00:47:29] Run 'pmbootstrap log' for details.
[00:47:29] See also: <https://postmarketos.org/troubleshooting>
```
which makes sense, since there is no boot.img file in the boot folder of the samsung-n7100's chroot.

**Comment by ollieparanoid** [comment]
As discussed with @drebrez: merging into an extra branch, so we can rebase it for the aports subfolder change. Thanks for your hard work on this, both @JamesHealdUK and @drebrez, and everyone else who helped! :+1:

**Comment by JamesHealdUK** [comment]
Thanks for the heads up both! :) Just fixing it up my side and will await your instruction on what to do with the merge

**Comment by ollieparanoid** [comment]
Excellent! Here are the instructions as I see it (be aware that I am not a git pro, but from reading [this](https://help.github.com/articles/syncing-a-fork/#platform-linux) article, I think it should work like the following):
* In a terminal, go to the folder, where you have your own repository checked out
* run `git fetch upstream` to get the latest source from our upstream `pmbootstrap` repo
* run `git merge upstream/device-samsung-n7100` to get the change, that moved your device- and linux- package into the new `aports/device` folder
* apply your fixes
* run `git commit`
* run `git push`
* create a new PR

As always, ask in the chat or here if you need any help.

**Comment by JamesHealdUK** [comment]
Thanks @ollieparanoid ! Submitted a new PR #293

**Pull request merged**